java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/RuntimeImageBuildStep.java

```diff
@@ -72,27 +74,27 @@ private String getBaseRuntimeImage(BuildContext buildContext, Artifact artifact)
     RuntimeConfig runtimeConfig = buildContext.getRuntimeConfig();
 
     // Check if the user has explicitly selected the compat runtime
-    if (buildContext.isCompatEnabled() && artifact.getType() != EXPLODED_WAR) {
-      throw new BuildStepException(String.format("App Engine APIs have been enabled. In order to "
-          + "use App Engine APIs, an exploded WAR artifact is required, but a %s artifact was "
-          + "found. See https://cloud.google.com/appengine/docs/flexible/java/upgrading for more "
-          + "detail.", artifact.getType()));
+    if (buildContext.isCompatEnabled()) {
+      if (artifact.getType() != EXPLODED_WAR) {
+        throw new BuildStepException(String.format("App Engine APIs have been enabled. In order to "
+            + "use App Engine APIs, an exploded WAR artifact is required, but a %s artifact was "
+            + "found. See https://cloud.google.com/appengine/docs/flexible/java/upgrading for more "
+            + "detail.", artifact.getType()));
+      }
+      return compatImageName;
     }
 
     // Select runtime based on artifact type
 
-    if (artifact.getType() == EXPLODED_WAR) {
-      // Use the compat runtime for exploded war artifacts.
-      logger.info("Using base image '{}' for exploded WAR artifact", compatImageName);
-      return compatImageName;
+    ArtifactType artifactType = artifact.getType();
 
-    } else if (artifact.getType() == WAR) {
+    if (artifactType == EXPLODED_WAR || artifactType == WAR) {
+      // Use the compat runtime for exploded war artifacts.
       String baseImage
           = jdkServerLookup.lookupServerImage(runtimeConfig.getJdk(), runtimeConfig.getServer());
-      logger.info("Using base image '{}' for WAR artifact", baseImage);
+      logger.info("Using base image '{}' for {} artifact", baseImage, artifactType);
```

Comment:
Put this at the end of the method, right before the return. You can then also avoid the duplication of this statement.